crates/polars-stream/src/morsel.rs

```diff
@@ -23,6 +23,15 @@ impl MorselSeq {
     pub fn new(seq: u64) -> Self {
         Self(seq.checked_mul(2).unwrap())
     }
+
+    // The morsel sequence id which comes after this morsel.
+    pub fn successor(self) -> Self {
+        Self(self.0.checked_add(2).unwrap())
```

Comment:
Can we have a comment why we have increments of 2?

Reply:
That's what the comment on `new` is for, the idea is to indicate the 'last morsel with this sequence id' using the final bit, so the sequence 1, 2, 3, 3, 3, 4 can become 2, 4, 6, 6, 7, 8 and as soon as we see 7 we know there won't be any 6s or 7s anymore.